srtcore/api.cpp

```diff
@@ -1919,36 +2084,65 @@ int CUDT::bind(SRTSOCKET u, const sockaddr* name, int namelen)
            // too small for particular family, or that family is
            // not recognized (is none of AF_INET, AF_INET6).
            // This is a user error.
-           s_UDTUnited.setError(new CUDTException(MJ_NOTSUP, MN_INVAL, 0));
-           return ERROR;
+           return setError(MJ_NOTSUP, MN_INVAL, 0);
        }
-       return s_UDTUnited.bind(u, sa);
+       CUDTSocket* s = s_UDTUnited.locateSocket(u);
```

Comment:
The `CUDTUnited::bind` has been changed to get the `CUDTSocket*` instead of `SRTSOCKET` and the socket lookup part was moved out of this. This is because this function will get much wider use, including a situation when the `CUDTSocket` object is already at hand.